enumerables.rb

```diff
@@ -0,0 +1,160 @@
+module Enumerable
+
+  # my_each
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_each
+    return to_enum(:my_each) unless block_given?
+      i = 0
+      while i < self.size do
+        yield(self.to_a[i])
+        i += 1
+      end
+  end
+  
+  # my_each_with_index
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+    i = 0
+    while i < self.length do
+      yield(i,self[i])
+      i += 1
+    end
+  end
+
+  # my_select
+  
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_select
+    return to_enum(:my_select) unless block_given?
+    arr = []
+    i = 0
+    while i < self.length do
+      arr << self[i] if yield(self[i])
+      i += 1
+    end
+    arr
+  end
+  
+ # my all
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_all?
+    unless block_given?
+      i = 0
+      while i < self.length do
+        return false unless self[i]
+        i += 1
+      end
+      return true
+    end
+    bool = true
+    i = 0
+    while i < self.length do
+      bool = false unless yield(self[i]) 
+      i += 1
+    end
+    bool
+  end
+
+ # my any
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_any?
+    unless block_given?
+      i = 0
+      while i < self.length do
+        return true unless self[i]
```

Comment:
I think this should be `if` instead of `unless`, you want to return true **if any** of the collection member is **truthy**. 

Please verify this. Also, see previous comments about accepting arguments. Also, check examples on ruby doc.